Library/Homebrew/cmd/uninstall.rb

```diff
@@ -78,28 +76,47 @@ def uninstall
     end
   end
 
-  def should_check_for_dependents?
-    # --ignore-dependencies, to be consistent with install
-    return false if ARGV.include?("--ignore-dependencies")
-    return false if ARGV.homebrew_developer?
-    true
+  def handle_unsatisfied_dependents(kegs_by_rack)
+    return if ARGV.include?("--ignore-dependencies")
+
+    all_kegs = kegs_by_rack.values.flatten(1)
+    check_for_dependents all_kegs
   end
 
   def check_for_dependents(kegs)
     return false unless result = Keg.find_some_installed_dependents(kegs)
 
-    requireds, dependents = result
+    if ARGV.homebrew_developer?
+      dependents_output_for_developers(*result)
+    else
+      dependents_output_for_nondevelopers(*result)
+    end
+
+    true
+  end
 
+  def dependents_output_for_developers(requireds, dependents)
+    msg = requireds.join(", ")
+    msg << (requireds.count == 1 ? " is" : " are")
+    msg << " required by #{dependents.join(", ")}, which "
+    msg << (dependents.count == 1 ? "is" : "are")
+    msg << " currently installed."
+    msg << "\nYou can silence this warning with "
+    msg << "`brew uninstall --ignore-dependencies "
+    msg << "#{requireds.map(&:name).join(" ")}`."
```

Comment:
Will merge this as-is but would be nice to refactor this to use `EOS.undent` and a bunch of variables declared before instead. Similarly below.